src/tools.ts

```diff
@@ -0,0 +1,60 @@
+import * as debugAgent from '@google-cloud/debug-agent';
+import * as profiler from '@google-cloud/profiler';
+import * as trace from '@google-cloud/trace-agent';
+
+let toolsAreSetup = false;
+
+/**
+ * Starts Google Cloud tooling.
+ * - StackDriver Trace
+ * - StackDriver Profiler
+ * - StackDriver Debugger
+ */
+export function startTooling() {
+  // Ensure tools are only setup once per process.
+  if (toolsAreSetup) {
+    return null;
+  } else {
+    toolsAreSetup = true;
+  }
+
+  // Start Trace
+  trace.start();
```

Comment:
I thought the trace-agent needed to be started as the very first part of the app:
```
require('@google-cloud/trace-agent').start();
```
See https://www.npmjs.com/package/@google-cloud/trace-agent#usage